src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1179,6 +1179,22 @@ def filter(self, col: Expression) -> "SelectStatement":
                 from_=self.to_subqueryable(), where=col, analyzer=self.analyzer
             )
 
+        # extract any subquery plan from ScalarSubquery expression that may be
+        # present in the col expression.
+        self.analyzer.subquery_plans = []
```

Comment:
@sfc-gh-aalam is that fixing for https://snowflakecomputing.atlassian.net/browse/SNOW-1678419 ?

Reply:
I guess it would fix that bug too. I discovered this when I was writing my in expression test for large query breakdown

Reply:
can yo search for the ticket number in the code base and enable the test to see if they start passing